### ISSUE TITLE
#[must_use] does not warn for unused must_use tuple components

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#26291]
``` rust
#[must_use]
struct MU;

fn a() -> (u8, MU) {
    (0, MU)
}

pub fn main() {
    a(); // does not warn
    a().1; // warns
    a().0; // does not warn
}
```

I believe it should warn in all three cases.

**Comment by steveklabnik** [comment]
Triage: no change

**Comment by Mark-Simulacrum** [comment]
Closing in favor of #39524.
